src/PendingReasonCron.php

```diff
@@ -113,71 +113,61 @@ public static function cronPendingreason_autobump_autosolve(CronTask $task)
                 continue;
             }
 
+            // Load pending reason
+            $pending_reason = PendingReason::getById($pending_item->fields['pendingreasons_id']);
+            if (!$pending_reason) {
+                trigger_error("Failed to load PendingReason", E_USER_WARNING);
+                continue;
+            }
+
+            // Skip if the day is a non-working day or a holliday
+            $calendar = Calendar::getById($pending_reason->fields['calendars_id']);
+            if ($calendar && ($calendar->isHoliday($now) || !$calendar->isAWorkingDay(time()))) {
+                continue;
+            }
```

Comment:
Is this check needed ?
If I understand correctly, `getNextFollowupDate` and `getAutoResolvedate` already take the calendar into account.